be/src/pipeline/exec/partitioned_aggregation_source_operator.cpp

```diff
@@ -133,13 +133,16 @@ Status PartitionedAggSourceOperatorX::get_block(RuntimeState* state, vectorized:
     SCOPED_TIMER(local_state.exec_time_counter());
     RETURN_IF_ERROR(local_state._status);
 
-    RETURN_IF_ERROR(local_state.initiate_merge_spill_partition_agg_data(state));
+    if (local_state._shared_state->is_spilled) {
+        RETURN_IF_ERROR(local_state.initiate_merge_spill_partition_agg_data(state));
```

Comment:
This method will merge block again. Do we need clear the previous partition data??

Reply:
`initiate_merge_spill_partition_agg_data` will check whether the current partition still has data, if it is, it continue to output data from the current partition; if not, will merge the next partition.